src/TesApi.Web/GroupableSet.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace TesApi.Web
+{
+    /// <summary>
+    /// Represents a set of values.
+    /// </summary>
+    /// <typeparam name="T">The type of elements in the hash set.</typeparam>
+    public class GroupableSet<T> : ISet<T>

```

Comment:
Seems a bit risky to write our own type like this with extremely difficult-to-diagnose potential bugs...  truly needed?

Reply:
See comment on KeyedGroup below